pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -553,6 +526,41 @@ func ResolvePipelineRunTask(
 	return &rprt, nil
 }
 
+func resolveTask(ctx context.Context, taskRun *v1beta1.TaskRun, getTask resources.GetTask, task v1beta1.PipelineTask) (v1beta1.TaskSpec, string, v1beta1.TaskKind, error) {
```

Comment:
thank you @jerop for refactoring 🙏 

not a blocker (its merged already 🤣) or the intent is not to demotivate but its always great to have a unit test for any new function introduced. We are carrying a lot of such functions without any unit tests since long time back.